### DISCUSSION TITLE
Could isMatching provide info about why (and where) if it did not match?

### DISCUSSION THREAD

**Discussion by mrft** [gvergnaud/ts-pattern#152] [Ideas]
If you want to use isMatching to validate external input (API, config file, ...), it would be useful to be able to give feedback about why the input was not acceptable.

Currently isMatching(...) only returns true or false, but it would be nice if it would return more info.

Does that seem like a good idea @gvergnaud?

**Comment by gvergnaud** [comment]
It would be great to have this feature, but probably as a separate helper from `isMatching`. `IsMatching` has to return a boolean to be a valid type predicate that can narrow your input type.

**Comment by nkoehring** [comment]
I think this flow would be pretty okay, even if means, that the matcher runs twice.

```ts
if(isMatching(obj)) return success(obj)
else return failure(matchDetails(obj, pattern))
```
